http_ingress_bridges/src/main/java/swim/forex/FreeForexApi.java

```diff
@@ -0,0 +1,102 @@
+// Copyright 2015-2019 SWIM.AI inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package swim.forex;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import javax.net.ssl.HttpsURLConnection;
+import swim.api.agent.AgentContext;
+import swim.json.Json;
+import swim.structure.Item;
+import swim.structure.Value;
+
+public final class FreeForexApi {
+
+  private HttpsURLConnection conn;
+
+  private FreeForexApi() {
+  }
+
+  private void openConn() {
+    if (this.conn != null) {
+      closeConn();
+    }
+    try {
+      final URL url = new URL(FREE_FOREX_API_URI);
+      this.conn = (HttpsURLConnection) url.openConnection();
+      this.conn.setDoOutput(true);
+      this.conn.setConnectTimeout(5000);
+      this.conn.setRequestMethod("GET");
+    } catch (IOException e) {
+      closeConn();
+      throw new RuntimeException("Failed to open connection");
+    }
+  }
+
+  private void closeConn() {
+    if (this.conn != null) {
+      this.conn.disconnect();
+      this.conn = null;
+    }
+  }
+
+  private Value fetchExchangeRates() throws IOException {
+    try (InputStreamReader is = new InputStreamReader(this.conn.getInputStream());
+         BufferedReader br = new BufferedReader(is)) {
+      return Json.parse(br.readLine());
+    }
+  }
+
+  public static void relayExchangeRates(AgentContext swim) throws IOException {
```

Comment:
Code flow contains PRIVATE-PUBLIC-PRIVATE. Can you move this below after all private flow. Reason being easier flow of the code to read.

Reply:
In this case it does indeed flow better that way. Thank you; I'm happy to make the change. In general though, it seems that the [code coventions](https://www.oracle.com/java/technologies/javase/codeconventions-fileorganization.html#1852) suggest to prioritize functionality:

> These methods should be grouped by functionality rather than by scope or accessibility. For example, a private class method can be in between two public instance methods. The goal is to make reading and understanding the code easier.